src/entities/FactorGradients.jl

```diff
@@ -58,7 +58,7 @@ Related
 
 [`calcFactorResidualTemporary`](@ref), [`_buildGraphByFactorAndTypes`](@ref)
 """
-mutable struct FactorGradientsCached!{F <: AbstractRelative,S,M<:Tuple,P,G,L}
+mutable struct FactorGradientsCached!{F <: AbstractRelative, S, M <: Vector{<:Tuple}, P, G, L}
```

Comment:
can perhaps move to `M <: AbstractVector{<:Tuple}` later, just leaving a breadcrum.